curvefs/src/client/curve_fuse_op.cpp

```diff
@@ -100,22 +127,44 @@ int InitFuseClient(const char *confPath, const char *fsType) {
 
     conf.PrintConfig();
 
+    g_fuseClientOption = new FuseClientOption();
+    curvefs::client::common::InitFuseClientOption(&conf, g_fuseClientOption);
+
+    std::shared_ptr<FsInfo> fsInfo = std::make_shared<FsInfo>();
+    if (GetFsInfo(fsName, fsInfo) != 0) {
+        return -1;
+    }
+
     std::string fsTypeStr = (fsType == nullptr) ? "" : fsType;
+    std::string fsTypeMds;
+    if (fsInfo->fstype() == FSType::TYPE_S3) {
+       fsTypeMds = "s3";
+    } else if (fsInfo->fstype() == FSType::TYPE_VOLUME) {
+       fsTypeMds = "volume";
+    }
 
-    fuseClientOption = new FuseClientOption();
-    curvefs::client::common::InitFuseClientOption(&conf, fuseClientOption);
-    if (fsTypeStr == "s3") {
+    if (fsTypeMds != fsTypeStr) {
+        LOG(ERROR) << "The parameter fstype is inconsistent with mds!";
+        return -1;
+    } else if (fsTypeStr == "s3") {
         g_ClientInstance = new FuseS3Client();
+        // set fsS3Option
+        const auto& s3Info = fsInfo->detail().s3info();
+        ::curve::common::FsS3Option fsS3Option;
+        ::curvefs::client::common::S3Info2FsS3Option(s3Info, &fsS3Option);
+        SetFuseClientS3Option(g_fuseClientOption, fsS3Option);
```

Comment:
can you just put these in fuse_s3_client ?

Reply:
> can you just put these in fuse_s3_client ?

good,idea! fix